modules/parrot_drush/manifests/components/php/drush_alias.pp

```diff
@@ -0,0 +1,24 @@
+define parrot_drush::components::php::drush_alias (
+  $root,
+  $ensure = 'present',
+  $uri = $name,
+  $php_executable = undef,
+  $protocol = '',
+)
+{
+
+  # Create home directories for the users that will be using Drush.
+  file { '/home/vagrant/.drush':
```

Comment:
This will die if there are two drush aliases, as you can't manage the same resource twice.

Reply:
You need to factor this out into a class, and then include that class (you can include it as many times as you like).

Reply:
Ah yeah, I totally fixed that but forgot to push. I may not have moved it to the best place though, I just put it in the first thing up the 'call stack' that wasn't per-site.